contracts/AckToEarn.sol

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "hardhat/console.sol";
+
+contract AckToEarn is Ownable, ReentrancyGuard {
+    using Counters for Counters.Counter;
+
+    event NewBid(address indexed from, address indexed to, uint amount);
+    event ReclaimBid(address indexed bidder, uint amount);
+
+    Counters.Counter private bidIds;
+
+    struct Bid {
+        uint id;
+        address bidder;
+        address recipient;
+        uint amount;
+        uint timestamp;
+        string message;
+        bool claimed;
+        bool exists;
+    }
+
+    mapping(address => mapping(uint => Bid)) private bidderBids;
+    mapping(address => mapping(uint => Bid)) private recipientBids;
+
+    Bid[] public bids;
+
+    uint constant bidExpiryThreshold = 7 days;
```

Comment:
didn't realize we can just say `7 days` like this. TIL!

Reply:
Yes, Solidity Time Units are helpful: https://docs.soliditylang.org/en/v0.8.15/units-and-global-variables.html#time-units